fastapi/dependencies/utils.py

```diff
@@ -410,6 +441,15 @@ def is_coroutine_callable(call: Callable) -> bool:
             continue
         if sub_dependant.use_cache and sub_dependant.cache_key in dependency_cache:
             solved = dependency_cache[sub_dependant.cache_key]
+        elif inspect.isgeneratorfunction(call) or inspect.isasyncgenfunction(call):
```

Comment:
If I recall correctly, I noticed that this code gets called at request time for each request. But it seems to me like it should (theoretically) be possible to perform this logic when the dependency is *created*, which could remove a substantial number of request-time checks.

I definitely don't think this is worth addressing now, and it may be hard/impossible to address in a way that is compatible with useful features like the dependency-overrides provider, but it might be worth looking into at some point.

Reply:
Agreed! With both comments :smile: